### ISSUE TITLE
consider using SE's API for getting all sites

### ISSUE THREAD

**Issue by moates** [ShivanKaul/SidebarOverflow#2]
Consider adding [this simple API](https://api.stackexchange.com/docs/sites#pagesize=999&filter=!.UDMsw2*Hs-oA4de&run=true) that pulls all SE's sites. That way new domains are covered for the future.

**Comment by ShivanKaul** [comment]
Hi! That's a great idea - would you be interested in working on it and perhaps opening a Pull Request? Else I'll give it a look sometime after the holidays :)

**Comment by moates** [comment]
(Sorry, accidentally closed this issue earlier)

I went to check this out (over a year later lol). Since the manifest only allows for hard-coded matching, to make the sites list dynamic, we'd have to request new permissions and complicate things. Given how rarely new sites are added, it's probably not worth it. But I pulled the most recent list of sites from the [SE API](https://api.stackexchange.com/docs/sites#pagesize=999&filter=!6P.EhvygxUiNX&run=true) and manually added them in #5 

For future reference, a lazy way of checking for new sites: 

`wget -O - 'https://api.stackexchange.com/2.2/sites?pagesize=999&filter=!6P.EhvygxUiNX' | gunzip -c > sites.txt'`

`sed 's/,/\n/g' sites.txt | grep -Ev '(stackexchange.com|stackoverflow.com)'  | grep 'http' | cut -d'"' -f4`
